초기 환경 구성

Kim-HJ1986 opened the pull request:
- 스프링 부트 프로젝트 생성
- 디렉토리 구조 생성
- kafka, redis, mysql -> local docker 컨테이너와 연동 테스트 샘플 코드 작성
- jpa 초기 세팅 -> model 디렉토리에 Entity 클래스 생성 후 워크벤치에서 확인

Kim-HJ1986 committed:
초기 환경 구성

kafka, redis 연동 테스트 및 디렉토리 구성

Kim-HJ1986 committed:
Remove build directory from tracking

Kim-HJ1986 committed:
Remove application.properties from Git tracking

Kim-HJ1986 committed:
초기 환경 구성

Kim-HJ1986 committed:
mysql 관련 초기 설정

jpa table model 생성 전략 설정
커스텀 timestamped 클래스 설정 (created_at, modified_at)

Kim-HJ1986 committed:
Remove .idea from Git tracking

jinspark-lab commented on src/main/java/com/project/watermelon/KafkaSampleConsumer.java:
```diff
@@ -0,0 +1,66 @@
+
+public class KafkaSampleConsumer {
+
+    public static void main(String[] args) {
```
Consumer Java Application 이랑 Producer Java Application 을 각각 경로 지정해서 따로 띄우실 거죠?
패키지 또는 모듈을 나누는게 좋을 것 같습니다.

Kim-HJ1986 replied:
넵 맞습니다, Consumer Java Application은 따로 구성해보겠습니다!

Kim-HJ1986 replied:
main-server, consumer-server 모듈을 분리하여 구성해두었습니다~!

jinspark-lab commented on src/main/java/com/project/watermelon/KafkaSampleConsumer.java:
```diff
@@ -0,0 +1,66 @@
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
```
자체를 앱으로 구동하려하신다면, property 를 따로 관리하고 불러오게 하는 편이 좋습니다.

Kim-HJ1986 replied:
넵 이부분 관련해서 서칭해보겠습니다~!

jinspark-lab commented on src/main/java/com/project/watermelon/KafkaSampleConsumer.java:
```diff
@@ -0,0 +1,66 @@
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
```
일부러 NPE 를 내시는것 보다는 예외를 던지는게 나을것 같습니다.

jinspark-lab commented on src/main/java/com/project/watermelon/KafkaSampleProducer.java:
```diff
@@ -0,0 +1,67 @@
+        props.put(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, "12000");
+
+        try (KafkaProducer<String, String> producer = new KafkaProducer<>(props)) {
+            for (int i = 10; i < 20; i++) {
```
자체가 샘플 코드인가요? 숫자의 의미를 모르겠습니다.

Kim-HJ1986 replied:
넵 도커에 카프카 서버 띄운 후 프로듀싱, 컨슈밍 테스트 해보는 코드였습니다. 
gitignore 에 등록해두겠습니다!

jinspark-lab commented on src/main/java/com/project/watermelon/RedisSampleService.java:
```diff
@@ -0,0 +1,33 @@
+        // 티켓 정보를 일정 시간 후 만료되도록 Redis에 저장
+        redisUtil.saveMessageWithExpiration(ticketId, ticketDetails, timeout, TimeUnit.HOURS);
+    }
+
```
샘플 로직인거죠? 로직이 좀 다른것 같아서 문의드립니다.

Kim-HJ1986 replied:
넵 이부분도 도커에 레디스 서버 띄운 후 메세지 저장되는 부분 확인용이었습니다!

jinspark-lab commented on src/main/java/com/project/watermelon/WatermelonApplication.java:
```diff
@@ -0,0 +1,29 @@
+@SpringBootApplication
+public class WatermelonApplication {
+
+	public static void main(String[] args) {
```
이 어플리케이션은 테스트 용도인가요?

Kim-HJ1986 replied:
해당 어플리케이션이 티켓팅 메인 웹 백엔드 서버가 될 예정입니다~!

jinspark-lab commented on src/main/java/com/project/watermelon/model/Performance.java:
```diff
@@ -0,0 +1,41 @@
+    private String endTime;
+
+    @Column(nullable = false)
+    private String locationId;
```
어떤 목적의 테이블인가요? 설계를 먼저 공유주시면 좋을 것 같습니다.

Kim-HJ1986 replied:
넵 ERD 먼저 슬랙으로 공유드리겠습니다~!

jinspark-lab commented on src/main/java/com/project/watermelon/util/RedisUtil.java:
```diff
@@ -0,0 +1,33 @@
+//    @Autowired
+//    public RedisUtil(RedisTemplate<String, Object> redisTemplate) {
+//        this.redisTemplate = redisTemplate;
+//    }
```
주석은 개발 때만 남겨주시기를 권장드립니다.

Kim-HJ1986 replied:
넵 !

jinspark-lab commented on src/test/java/com/project/watermelon/WatermelonApplicationTests.java:
```diff
@@ -0,0 +1,13 @@
+
+	@Test
+	void contextLoads() {
+	}
```
빈 테스트 코드는 지워주세요.

Kim-HJ1986 replied:
넵 지우겠습니다.

jinspark-lab reviewed (commented):
피드백 드렸습니다. 예제 코드들은 가급적 PR 반영 안하시는 것이 좋습니다.
환경설정이 마무리가 안된것 같아서, 그 부분 한번 신경써보시면 좋을것 같습니다.

Kim-HJ1986 committed:
main-server / consumer-server 모듈 분리

main-server / consumer-server 모듈 분리
- main-server: main ticketing was server
- consumer-server: kafka consumer server

Kim-HJ1986 committed:
redis sample test code 삭제

샘플 테스트 코드 삭제 후
초기화 변수 설정 ( 최대 가용 카운트, 현재 진행 카운트 수 설정)

Kim-HJ1986 committed:
kafka consumer server api 추가

카프카 컨슈머 서버 API 추가
- 특정 오프셋 부터 조회해오도록 commit 설정 및 seek 메서드 사용

Kim-HJ1986 committed:
main-server ticket 가용성 체크 API 추가

Kim-HJ1986 committed:
변수명 수정

Kim-HJ1986 committed:
ticket waiting message produce API 추가

Kim-HJ1986 committed:
샘플코드 삭제, 콘서트 테이블 변경 사항 적용

PERFORMANCE -> CONCERT 로 테이블 명 변경
CONCERT_ID의 경우 auto increment로 관리

Kim-HJ1986 committed:
빈 테스트 코드 삭제

Kim-HJ1986 committed:
kafka listener annotation 구조 구현
